src/components/Expense/CreateExpense.jsx

```diff
@@ -0,0 +1,227 @@
+import { useState, useContext, useEffect } from "react";
+import { collection, addDoc } from "firebase/firestore";
+
+import { db } from "../../firebase-config";
+import AddBorrower from "./AddBorrower";
+import { userContext } from "../../App";
+import Input from "../Elements/Input";
+
+import "./Expense.css";
+
+const CreateExpense = () => {
+  const [data, setData] = useState({ description: "", totalBill: 0 });
+
+  const [payerId, setPayerId] = useState("");
+  const [payerBill, setPayerBill] = useState(0);
+
+  const [borrowerTotalBill, setBorrowerTotalBill] = useState(0);
+  const [borrowers, setBorrowers] = useState([]);
+  const { users } = useContext(userContext);
+
+  const changeHandler = (event) => {
+    const { name, value } = event.target;
+    setData({ ...data, [name]: value });
+  };
+
+  const canSubmitFirebase = () => {
+    if (data.description === "") {
+      alert("Description can not be empty");
+    } else if (data.totalBill <= 0 || data.totalBill === "") {
+      alert("Total Bill Invalid");
+    } else if (payerId === "") {
+      alert("Please Select Paid by");
+    } else if (+data.totalBill !== borrowerTotalBill) {
+      alert("Total Bill shuold be equal to Shared Bill");
+    } else {
+      console.log("Local checks passed.");
+      setDataFirebase();
+    }
+  };
+
+  const canBeSubmitFirebase = () => {
```

Comment:
Can we not update this method and above one as they didn't look so much great with so many checks
Kindly don't take it on mind